### ISSUE TITLE
KeyNotFoundException in NotificationReceiver.OnReceive

### ISSUE THREAD

**Issue by erichedstrom** [EgorBo/Toasts.Forms.Plugin#134]
I haven't seen this locally, but have gotten a few crash reports with this stack trace:

```
Xamarin Exception Stack:
System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <30c2c7f3246e4172a1fa8435697a5fe7>:0
  at Plugin.Toasts.NotificationReceiver.OnReceive (Android.Content.Context context, Android.Content.Intent intent) [0x0006c] in <b81182804c7e489493d406b6d34a5576>:0
  at Android.Content.BroadcastReceiver.n_OnReceive_Landroid_content_Context_Landroid_content_Intent_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_context, System.IntPtr native_intent) [0x00017] in <0bff0d33fcda4ec097a7b9046bee9937>:0
  at (wrapper dynamic-method) System.Object:ef788143-1450-4f91-a6fd-72f04a092915 (intptr,intptr,intptr,intptr)
```

I am using version 3.2.1 of the plugin with Xamarin.Forms 2.3.4.270.

**Comment by mgoodfellow** [comment]
Hi @erichedstrom,

I have been suffering this same error actually. Just wanted to ask if you are awaiting the call to Notify? `await notification.Notify(options)`

I haven't been and I sometimes fall foul of this error. I have just built a new version of my app with awaits in place and I will report back if that solves the issue.

Just having a look at the Droid codebase and it appears that it is [this line](https://github.com/EgorBo/Toasts.Forms.Plugin/blob/master/Toasts.Forms.Plugin.Droid/NotificationBuilder.cs#L228) here which throws the exception.

Obviously we could check for the existence of the key before trying to call `.Set()`. It worries me that something should be waiting on the event though, as it is waiting in the code for [Notify](https://github.com/EgorBo/Toasts.Forms.Plugin/blob/master/Toasts.Forms.Plugin.Droid/NotificationBuilder.cs#L166)

**Comment by adamped** [comment]
Fixed in 3.3.0-pre1
